src/core/namespace.c

```diff
@@ -1173,8 +1173,7 @@ int setup_namespace(
 
         if (root_directory)
                 root = root_directory;
-        else if (root_image || n_bind_mounts > 0 || n_temporary_filesystems > 0) {
-
+        else {
                 /* If we are booting from an image, create a mount point for the image, if it's still missing. We use
                  * the same mount point for all images, which is safe, since they all live in their own namespaces
                  * after all, and hence won't see each other. We also use such a root directory whenever there are bind
```

Comment:
this comment needs some updating. It still suggests that this code only applies if we boot from an image. I figure you can drop much of the comment now. But it would be best if you'd explain here why we are not using the host's root here ever